src/access/acp.ts

```diff
@@ -715,6 +715,157 @@ export function internal_setActorAccess<
+ * Policies specify that everybody is *denied* Read access *except* if they're
+ * in a particular Group, then that will be left intact.
+ * This means that, unless *only* this module's functions are used to manipulate
+ * access to this Resource, the set access might not be equal to the effective
+ * access for a particular Agent.
+ *
+ * There are a number of preconditions that have to be fulfilled for this
+ * function to work:
+ * - Access to the Resource is determined via an Access Control Resource.
+ * - The Resource's Access Control Resource does not refer to (Policies or Rules
+ *   in) other Resources.
+ * - The current user has access to the Resource's Access Control Resource.
+ *
+ * If those conditions do not hold, this function will return `null`.
+ *
+ * Additionally, take note that the given access will only be applied to the
+ * given Resource; if that Resource is a Container, access will have to be set
+ * for its children independently.
+ *
+ * @param resource Resource that was fetched together with its linked Access Control Resource.
+ * @param access What access (read, append, write, controlRead, controlWrite) to set for everybody. `true` to allow, `false` to deny, and `undefined` to leave unchanged.
+ * @returns The Resource with the updated Access Control Resource attached, if updated successfully, or `null` if not.
+ */
+export function internal_setPublicAccess<
+  ResourceExt extends WithResourceInfo & WithAcp
+>(resource: ResourceExt, access: Access): ResourceExt | null {
+  return internal_setActorAccess(resource, acp.agent, acp.PublicAgent, access);
+}
+
+/**
+ * Set access for a authenticated Agents to a Resource.
+ *
+ * This function adds the relevant Access Control Policies and Rules to a
+ * Resource's Access Control Resource to define the given access for
+ * authenticated Agents specifically. In other words, it can, for example, add
+ * Policies that give authenticated Agents Read access to the Resource. However,
+ * if other Policies specify that authenaticated Agents are *denied* Read access
+ * *except* if they're in a particular Group, then that will be left intact.
+ * This means that, unless *only* this module's functions are used to manipulate
+ * access to this Resource, the set access might not be equal to the effective
+ * access for a particular Agent.
+ *
+ * There are a number of preconditions that have to be fulfilled for this
+ * function to work:
+ * - Access to the Resource is determined via an Access Control Resource.
+ * - The Resource's Access Control Resource does not refer to (Policies or Rules
+ *   in) other Resources.
+ * - The current user has access to the Resource's Access Control Resource.
+ *
+ * If those conditions do not hold, this function will return `null`.
+ *
+ * Additionally, take note that the given access will only be applied to the
+ * given Resource; if that Resource is a Container, access will have to be set
+ * for its children independently.
```

Comment:
I guess you've used the term 'children' consistently everywhere, but I'd have used 'contained resources' instead (it just seems more technically precise to me):
```suggestion
 * for its contained resources independently.
```

Reply:
61891be3